docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,342 @@
+
+**Operation**
+
+Pseudocode:
+```rust
+loop {
+  let signed: SignedStatement = network_statements.next()?;
+  if let Statement::Seconded(candidate) = signed.statement {
+    if candidate is unknown and in local assignment {
+      spawn_validation_work(candidate);
+    }
+  }
+
+  // Track which candidates are now backed and watch out for double-voting misbehavior.
+  track_statement(signed);
+}
+```
+
+```rust
+fn spawn_validation_work(candidate) {
+  asynchronously {
+    let pov = network.fetch_pov(candidate).await;
+    let valid = validate_candidate(candidate, validation_function, parachain_head, pov);
+    if valid {
+      // make PoV available for later distribution.
+      // sign and dispatch `valid` statement to network.
+    } else {
+      // sign and dispatch `invalid` statement to network.
+    }
+  }
+}
+```
+
+[TODO: how to dispatch `Seconded` message. Kind of hard - note that dispatching a `Seconded` and `Valid` message is illegal so this is racy]
+
+### Candidate Proposal Process
+
+[TODO: get candidate from collator, feed to candidate backing process. ]
+
+### Secondary Checking
+
+[TODO]
+
+### Validator Assignment Process
+
+[TODO]
+
+### Collation Distribution Process
+
+[TODO]
+
+### Availability Distribution Process
+
+[TODO]
+
+----
+
+## Data Structures and Types
+
+[TODO]
+* CandidateReceipt
+* CandidateCommitments
+* AbridgedCandidateReceipt
+* GlobalValidationSchedule
+* LocalValidationData
+
+#### Block Import Event
+```rust
+/// Indicates that a new block has been added to the blockchain.
+struct BlockImportEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header itself.
+  header: Header,
+  /// Whether this block is considered the head of the best chain according to the 
+  /// event emitter's fork-choice rule.
+  new_best: bool,
+}
+```
+
+#### Block Finalization Event
+```rust
+/// Indicates that a new block has been finalized.
+struct BlockFinalizationEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header of the finalized block.
+  header: Header,
+}
+```
+
+#### Statement Type
+```rust
+/// A statement about the validity of a parachain candidate.
+enum Statement {
+  /// A statement about a new candidate.
+  Seconded(CandidateReceipt),
+  /// A statement about the validity of a candidate, based on candidate's hash.
+  Valid(Hash),
+  /// A statement about the invalidity of a candidate.
+  Invalid(Hash),
+}
+```
+
+#### Signed Statement Type
+
+This is a signed statement. The actual signed payload should reference only the hash of the CandidateReceipt and should include
+a relay parent which provides context to the signature. This prevents against replay attacks and allows the candidate receipt itself
+to be omitted when checking a signature on a `Seconded` statement.
+
+```rust
+/// A signed statement.
+struct SignedStatement {
+  statement: Statement,
+  signed: ValidatorId,
+  signature: Signature
+}
+```
+
+----
+
+## Glossary
```

Comment:
Why not link and extend https://substrate.dev/docs/en/overview/glossary ? (Why yet another Glossary ? :yum: )

Reply:
I'll add the link, but it doesn't seem like that glossary includes all of the terms I've been using, some of which are new. That's also a "Substrate" glossary, as opposed to a Polkadot one. cc @joshorndorff - I'll keep the terms here for now, but I'm hoping you can shed some light on the documentation story and where would be the best fit for terms that are mostly needed only to explain parachain-agreement internals.

Reply:
The Substrate now lives here https://www.substrate.io/kb/getting-started/glossary and the source is here https://github.com/substrate-developer-hub/knowledgebase/blob/master/current/getting-started/glossary.md

I agree we want to keep it a Substrate glossary and not introduce polkadot-specific terms. There is also the polkadot wiki's glossary which probably makes more sense in this case https://wiki.polkadot.network/docs/en/glossary

Reply:
Who's responsible for the Polkadot glossary? Is that on Web3's side?

Reply:
On Web3's side. I'm not sure who's responsible.